app/modules/somfy.c

```diff
@@ -0,0 +1,247 @@
+            DIRECT_WRITE_HIGH(pin);
+            signalindex++;
+            // delayMicroseconds(4*SYMBOL);
+            break;
+        case 4:
+            DIRECT_WRITE_LOW(pin);
+            subindex++;
+            if (subindex < sync) {signalindex--;} else {signalindex++;}
+            // delayMicroseconds(4*SYMBOL);
+            break;
+        case 5:
+            // Software sync
+            DIRECT_WRITE_HIGH(pin);
+            signalindex++;
+            // delayMicroseconds(4550);
+            break;
+        case 6:
+            DIRECT_WRITE_LOW(pin);
+            signalindex++;
+            subindex=0;
+            // delayMicroseconds(SYMBOL);
+            break;
+        case 7:
+            //Data: bits are sent one by one, starting with the MSB.
+            bitcondition = ((frame[subindex/8] >> (7 - (subindex%8))) & 1) == 1;
+            if(bitcondition) {
+                DIRECT_WRITE_LOW(pin);
+            }
+            else {
+                DIRECT_WRITE_HIGH(pin);
+            }
+            signalindex++;
+            // delayMicroseconds(SYMBOL);
+            break;
+        case 8:
+            //Data: bits are sent one by one, starting with the MSB.
+            if(bitcondition) {
+                DIRECT_WRITE_HIGH(pin);
+            }
+            else {
+                DIRECT_WRITE_LOW(pin);
+            }
+            
+            if (subindex<56) {
+                subindex++;
+                signalindex--;
+            }
+            else {
+                signalindex++;
+            }
+            // delayMicroseconds(SYMBOL);
+            break;
+        case 9:
+            DIRECT_WRITE_LOW(pin);
+            signalindex++;
+            // delayMicroseconds(30415); // Inter-frame silence
+            break;
+        case 10:
+            repeatindex++;
+            if (repeatindex<repeat) {
+                DIRECT_WRITE_HIGH(pin); //start repeat from step 3, but don't wait as after step 1
+                signalindex=4; subindex=0; sync=7;
+            } else {
+                platform_hw_timer_close(TIMER_OWNER);
+                if (lua_done_ref != LUA_NOREF) {
+                    task_post_low (done_taskid, (task_param_t)0);
+                }
+            }
+            break;
+    }
+    if (signalindex<10) {
+        platform_hw_timer_arm_ticks(TIMER_OWNER, delay[signalindex-1]);
+    }
+}
+
+static int somfy_lua_sendcommand(lua_State* L) { // pin, remote, command, rolling_code, num_repeat, callback
+    if (!lua_isnumber(L, 4)) {
+        return luaL_error(L, "wrong arg range");
+    }
+    pin = luaL_checkinteger(L, 1);
+    uint64_t remote = luaL_checkinteger(L, 2);
+    uint8_t cmd = luaL_checkinteger(L, 3);
+    uint16_t code = luaL_checkinteger(L, 4);
+    repeat=luaL_optint( L, 5, 2 );
+
+    luaL_argcheck(L, platform_gpio_exists(pin), 1, "Invalid pin");
+
+    if (!lua_isnoneornil(L, 6)) {
+        lua_pushvalue(L, 6);
+        lua_done_ref = luaL_ref(L, LUA_REGISTRYINDEX);
```

Comment:
You need to do luaL_unref(L, LUA_REGISTRYINDEX, lua_done_ref) first (just in case there is still a reference) (actually in both cases -- so do it a few more lines up).